docs/creating-sql-with-python.md

```diff
@@ -0,0 +1,201 @@
+If you are only worried about building counts tables, skip this section - 
+we've got enough wrappers that you shouldn't need to worry about this
+level of detail.
+
+For validating SQL, we are using 
+[Jinja templates](https://jinja.palletsprojects.com/en/3.1.x/)
+to create validated SQL in a repeatable manner. We don't expect you to write these
+templates - instead, using the 
+[template function library](../cumulus_library/template_sql/templates.py)
+you can provide a series of arguments to these templates that will allow you to
+generate standard types of SQL tables, as well as using templates targeted for
+bespoke operations. 
+
+When you're thinking about a query that you'd need to create, first check the
+template function library to see if something already exists. Basic CRUD
+should be covered, as well as unnestings for some common FHIR objects.
+
+## Use cases
+
+### Generating counts tables
+A thing we do over and over as part of studies is generate powerset counts tables
+against a filtered resource to get data about a certain kind of clinical population.
+Since this is so common we created a class just for this, and we're using it in all
+studies the Cumulus team is directly authoring.
+
+The [CountsBuilder class](../cumulus_library/schema/counts.py) 
+provides a number of convenience methods that are available for use (this covers
+mechanics of generation). You can see examples of usage in the 
+[Core counts builder](../cumulus_library/studies/core/count_core.py)
+(which is where the business logic of your study lives). 
+
+- `get_table_name` will scan the study's `manifest.toml` and auto prepend a table
+name with whatever the study prefix is.
+- `get_where_clauses` will format a string, or an array, of where clauses in a
+manner that the table constructors will expect.
+- `count_[condition,document,encounter,observation,patient]` will take a target table
+name, a source table, and an array of columns, and produce the appropriate powerset
+table to count that resource. You can optionally provide a list of where statements
+for filtering, or can change the minimum bin size used to include data
+- The `count_*` functions pass through to `get_count_query` - if you have a use
+case we're not covering, you can use this interface directly. We'd love to hear
+about it - we'd consider covering it and/or take PRs for new features
+
+As a convenience, if you include a `if __name__ == "__main__":` clause like you
+see in `count_core.py`, you can invoke the builder's output by invoking it with
+python, which is a nice way to get example SQL output for inclusion in github.
+This is where the 
+[count core sql output](../cumulus_library/studies/core/count_core.sql)
+originated from.
+
+Add your count generator file to the `counts_builder_config` section of your
+`manifest.toml` to include it in your build invocations.
+
+### Adding a static dataset
+
+*NOTE* - we have an
+[open issue](https://github.com/smart-on-fhir/cumulus-library/issues/58)
```

Comment:
You should make a note in that issue to come delete this line :)

Reply:
What, you don't trust my steel trap memory?


... what were we talking about